src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java

```diff
@@ -131,6 +132,52 @@ public void createTable(final String tableName) {
     createTable(tableName, false);
   }
 
+  public void createTableWithSchema(final String tableName, final Map<String, String> schema) {
+    if (schema.isEmpty()) {
+      createTableWithOnlyMetadataColumn(tableName);
+      return;
+    }
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query = "create table if not exists identifier(?)";
+    query += "(record_metadata variant";
+    String logColumn = "[RECORD_METADATA (VARIANT)";
+    for (Map.Entry<String, String> field : schema.entrySet()) {
+      query += ", " + field.getKey() + " " + field.getValue();
+      logColumn += ", " + field.getKey() + " (" + field.getValue() + ")";
+    }
+    query += ")";
+    logColumn += "]";
+
+    try {
+      PreparedStatement stmt = conn.prepareStatement(query);
```

Comment:
lets do a `logInfo` here too. In case of failures, we would want to know what was the query getting executed.

Reply:
first line of try catch